### PR TITLE
A Little Cleanup

### PR DESCRIPTION
**Pull request by CodaFi**
- Cleans up the docs a bit.
- Adds `invert` to `Iso`s.

### CODE REVIEW

File: Focus/Iso.swift [hunk 1]
```diff
@@ -6,12 +6,21 @@
...
 ///
-/// - parameter S: The source of the Iso heading right
-/// - parameter T: The target of the Iso heading left
-/// - parameter A: The source of the Iso heading right
-/// - parameter B: The target of the Iso heading left
+/// In practice, an Iso is used with two structures that can be converted between each other without
+/// information loss.  For example, the Isomorphism between `Optional<T>` and `CollectionOfOne<T?>`
+/// is expressed as
+///
+///     Iso<Optional<T>, Optional<U>, CollectionOfOne<T?>, CollectionOfOne<U>>
```
Comment (pthariensflame) [inline Focus/Iso.swift]: Should be `CollectionOfOne<U?>`, yes?

File: Focus/Iso.swift [hunk 2]
```diff
@@ -6,12 +6,21 @@
...
+///
+/// If a less-powerful form of `Iso` is needed, or `S == T` and `A == B`, consider using a
```
Comment (pthariensflame) [inline Focus/Iso.swift]: I think "or" should be "where" on this line.

File: Focus/Lens.swift [hunk 3]
```diff
@@ -6,12 +6,20 @@
...
 /// composing with other lenses to focus deeper into a structure, and returning new structures with
-/// parts modified.  In this way, a Lens can be thought of as a reference to a subpart of a
+/// parts modified.  In this way, a `Lens` can be thought of as a reference to a subpart of a
 /// structure.
 ///
-/// A well-behaved Lens should obey the following laws:
+/// In practice, a `Lens` is used with Product structures like tuples, classes, and structs. If a 
+/// less-powerful form of `Lens` is needed, consider using a `SimpleLens` instead.
+///
+/// A Lens, its is simplest form, can also be seen as a pair of functions:
```
Comment (pthariensflame) [inline Focus/Lens.swift]: "its is" should be "in its"

File: Focus/Lens.swift [hunk 4]
```diff
@@ -38,11 +46,12 @@ public struct Lens<S, T, A, B> : LensType {
 	/// Gets the Indexed Costate Comonad Coalgebroid underlying the receiver.
 	private let _run : S -> IxStore<A, B, T>
 
+	/// Runs the lens on a structure to retrieve the underlying Indexed Costate Comonad Coalgebroid.
 	public func run(v : S) -> IxStore<A, B, T> {
 		return _run(v)
 	}
 
-	/// Produces a lens from an Indexed Costate Comonad Coalgebroid.
+	/// Produces a lens from a function yielding an Indexed Costate Comonad Coalgebroid.
```
Comment (pthariensflame) [inline Focus/Lens.swift]: A coalgebra is _already_ a function yielding a particular functor (in this case, the indexed store/costate comonad).  This "correction" is strictly _in_correct!

File: Focus/Prism.swift [hunk 5]
```diff
@@ -6,7 +6,25 @@
 //  Copyright (c) 2015 TypeLift. All rights reserved.
 //
 
-/// A Prism is an `Iso` where one of the functions is partial.
+/// A `Prism` describes a way of focusing on potentially more than one target structure.  Like
+/// an `Iso` a `Prism` is invertible, but unlike an `Iso` multi-focus `Prism`s are incompatible with
+/// inversion in general.  Because of this a `Prism` describes a `Lens` that can branch depending on
```
Comment (pthariensflame) [inline Focus/Prism.swift]: A `Prism` is not a `Lens` (unless it is also an `Iso`).  This sentence needs rewording.

File: Focus/Prism.swift [hunk 6]
```diff
@@ -6,7 +6,25 @@
...
+/// whether it hits its target, much like pattern-matching in a `switch` statement.
+///
+/// An famous example of a `Prism` is
+///
+///     Prism<Optional<T>, Optional<T>, T, T>
+///
+/// provided by the `_Some` and `_None` `Prism`s in this library.
```
Comment (pthariensflame) [inline Focus/Prism.swift]: Only `_Some` has this type.  (Or so I hope!)

**Comment by pthariensflame** [comment]
Other than the noted complaints, looking good so far!  :)

File: Focus/Prism.swift (hunk 6 above)
Comment (CodaFi) [inline Focus/Prism.swift]: Uh... Fixed.

File: Focus/Iso.swift (hunk 1 above)
Comment (CodaFi) [inline Focus/Iso.swift]: Hm, yeah.  I'm gonna change the example anyways.  This makes a better `Prism` than it does an `Iso`.

File: Focus/Iso.swift [hunk 7]
```diff
@@ -6,12 +6,21 @@
...
 ///
-/// - parameter S: The source of the Iso heading right
-/// - parameter T: The target of the Iso heading left
-/// - parameter A: The source of the Iso heading right
-/// - parameter B: The target of the Iso heading left
+/// In practice, an `Iso` is used with two structures that can be converted between each other 
+/// without information loss.  For example, the Isomorphism between `Optional<T>` and 
+/// `ImplicitlyUnwrappedOptional<T>` is expressed as
+///
+///     Iso<Optional<T>, Optional<T>, ImplicitlyUnwrappedOptional<U>, ImplicitlyUnwrappedOptional<U>>
```
Comment (pthariensflame) [inline Focus/Iso.swift]: This is now wrong in a _different_ way.  Transpose the middle `T`-`U` pair; otherwise that type is uninhabitable (modulo bottom).

**Comment by CodaFi** [comment]
Ready for final review.

**Comment by pthariensflame** [comment]
LGTM!

**Comment by pthariensflame** [comment]
:boat:

**Pull request merged**